From c9b502bd7507cecb4e39bd3f021811a3ec50488e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:16:51 +0200
Subject: [PATCH 17/46] AnnaThermostat: add support for
 maximum_modulation_level

Parse and publish maximum_modulation_level as a percentage sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 3d851afcd..bc59d7a12 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1135,6 +1135,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendPercentageSensor(applianceIDhex, 2, batterypercentage, level, ApplianceName + " " + sname);;
                 }
             }
+			else if (sname == "maximum_modulation_level")
+			{
+				if (!tmpstr.empty())
+				{
+					float maxlevel = (float)atof(tmpstr.c_str()) * 100.F;
+					SendPercentageSensor(applianceIDhex, 3, batterypercentage, maxlevel, ApplianceName + " " + sname);
+				}
+			}
             else if (sname == "central_heater_water_pressure")
             {
                 if (!tmpstr.empty())
